From 894fd3a62f2ff6e702d6ff0e2a63182fa45ea216 Mon Sep 17 00:00:00 2001
From: Vlad-Andrei Badoiu <vlad_andrei.badoiu@stud.acs.upb.ro>
Date: Sat, 9 Nov 2019 00:33:52 +0200
Subject: [PATCH 1/1] Convert pte_handle_t to unsigned long

Signed-off-by: Vlad-Andrei Badoiu <vlad_andrei.badoiu@stud.acs.upb.ro>
---
 boost/thread/detail/thread.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/boost/thread/detail/thread.hpp b/boost/thread/detail/thread.hpp
index 04223ed26..2be0bc4c6 100644
--- a/boost/thread/detail/thread.hpp
+++ b/boost/thread/detail/thread.hpp
@@ -606,7 +606,7 @@ namespace boost
         hash_value(const thread::id &v)
         {
 #if defined BOOST_THREAD_PROVIDES_BASIC_THREAD_ID
-          return hash_value(v.thread_data);
+          return hash_value(*((unsigned long *)(pte_handle_t *)&v.thread_data));
 #else
           return hash_value(v.thread_data.get());
 #endif
-- 
2.20.1

